### PR TITLE
(Feature) robot communication and state

### PR DESCRIPTION
**Pull request by mcous**
Hey y'all. This PR takes a bunch of the robot communication work @astaff and I have been working on and wires it into the redux store. It also starts connecting that redux store to the application itself. Currently, we have (see `app/ui/containers/root` for the state to props mapping):

* isRunning -> true or false
* isConnected -> true or false
* connectionStatus -> 'connected', 'connecting', or 'disconnected'
* onRunButtonClick -> runs dinosaur.py when the run button is clicked

There's also some chore work wrapped up in here:

* JS coverage is go. From the `app` directory, run:
    * `make test-unit` - Runs unit tests once and calculates coverage
    * `make test-unit watch=true` - Runs unit test in watch mode without coverage
    * `make test-unit watch=true cover=true` - Runs tests in watch mode and calculates coverage on every run (this is a big output, hence why its not the default for watch mode)
* e2e tests have been COMMENTED OUT
    * They depend on a bunch of hooks that don't currently exist
    * We'll be bringing them back up very shortly now that we have one action wired up
* A bunch of boilerplate files have been removed

@Kadee80 and @IanLondon please review for general JS stuff and also that you're happy with how I'm laying out the application

@astaff I had a bad rebase (forgot to hit save during conflict resolution: I blame my text editor entirely because I am perfect). Can you please closely check the python side to make sure I put everything back in place properly? (Thank you for helping me with my incompetence)

### CODE REVIEW

**Comment by Kadee80** [comment]
Lgtm I'll reach out with any questions that arise with state.

**Comment by mcous** [comment]
Thanks @Kadee80, completely missed renaming the CSS file

File: api/opentrons/server/rpc.py [hunk 1]
```diff
@@ -53,38 +53,92 @@ def __init__(self, root=None, host='127.0.0.1', port=31950):
...
+        if id not in self.objects:
             raise ValueError(
-                'dispatch: object with id {0} not found'.format(_id))
+                'dispatch: object with id {0} not found'.format(id))
 
-        obj = self.objects[_id]
+        obj = self.objects[id]
         function = getattr(type(obj), name)
         args = self.resolve_args(args)
+        kwargs = {}
```
Comment (astaff) [inline api/opentrons/server/rpc.py]: Add comment with a note how tail argument is expanded into kwargs if it's a dict

File: api/opentrons/server/rpc.py [hunk 2]
```diff
@@ -121,94 +193,34 @@ def resolve(a):
                         'type': CALL_RESULT_MESSAGE
                     }
                 }
-                # Replace id with _id to be compatible with
+                # Replace id with id to be compatible with
```
Comment (astaff) [inline api/opentrons/server/rpc.py]: Nonsense. Will fix back to _id

File: app/test/e2e/integration_test.js [hunk 3]
```diff
@@ -1,127 +1,127 @@
-/* global describe, it, beforeEach, afterEach */
```
Comment (astaff) [inline app/test/e2e/integration_test.js]: @mcous not sure if these tests are of any particular use right now. @SimplyAhmazing WDYT?

**Comment by astaff** [comment]
Python code fixed as per review. LGTM 👍

File: app/ui/robot/api-client.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+// TODO(mc): as written, this is pretty hard to test. Spend some time thinking
+// about how to refactor so that this is not the case
+
+import assert from 'assert'
+import Client from '../../rpc/client'
+import {actions, actionTypes} from './'
+
+// TODO(mc): don't hardcode this URL or protocol
+const URL = 'ws://127.0.0.1:31950'
+const PROTOCOL = '/Users/mc/opentrons/opentrons/api/opentrons/server/tests/data/dinosaur.py'
```
Comment (mcous) [inline app/ui/robot/api-client.js]: Blah my bad for using a hardcoded absolute path; I should've never committed this. Fix this to a relative path

**Comment by mcous** [comment]
Closing in favor of #308

**Pull request closed without merging**